tests/kernels/test_causal_conv1d.py

```diff
@@ -271,6 +272,52 @@ def test_causal_conv1d_update(dim, width, seqlen, has_bias, silu_activation,
     ))
 
 
+@pytest.mark.parametrize("itype", [torch.bfloat16])
+@pytest.mark.parametrize("silu_activation", [True])
+@pytest.mark.parametrize("has_bias", [True])
+@pytest.mark.parametrize("seqlen", [1])
+@pytest.mark.parametrize("width", [4])
+@pytest.mark.parametrize("dim", [2048])
+def test_causal_conv1d_update_with_batch_gather_padding_unchanged(
+        dim, width, seqlen, has_bias, silu_activation, itype):
+    device = "cuda"
+    # set seed
+    seed_everything(0)
+    batch = 64
+
+    x = torch.randn(batch, dim, 1, device=device, dtype=itype)
+
+    total_entries = 10 * batch
+    conv_state = torch.randn(total_entries,
+                             dim,
+                             width - 1,
+                             device=device,
+                             dtype=itype)
+    conv_state_before = conv_state.clone()
+    conv_state_indices = torch.as_tensor([PAD_SLOT_ID] * batch,
+                                         dtype=torch.int32,
+                                         device=device)
```

Comment:
The test would be improved if it had some valid entries and some padded ones. Ditto for `test_causal_conv1d_varlen_check_padding_state_unchanged` and `test_selective_scan_varlen_padding_state_unchanged`

Reply:
I've removed those tests and embedded this option inside the existing tests, as a flag